Fixed version number error in setup.py and fixed CI on travis

Bounder opened the pull request:
# Description

These changes fix a non existing version in setup.py, which causes problems when installing this package using pipenv. Also CI building has been fixed and the project is now tested against 1.8 and 1.11

# Motivation

The original intent was fixing a not existing version number in setup.py which causes problems when installing this project using pipenv. However when forking the project I saw the CI was broken so I tried to fix that. I'm using this package in a lot of my project so it seemed logical to get this fixed and make sure it works in the python versions that are currently mainstream.

Bounder committed:
Fixed mistake in django version requirement

Bounder committed:
Increased max-line-length(flake)

Bounder committed:
Adjusted tests for Django 1.11

Bounder committed:
Only test for LTS releases

Bounder committed:
Fixed style error

Bounder committed:
Fixed travis configuration

Bounder committed:
Added travis testing of python 3.6

Bounder committed:
Fixed missing apt dependency in travis

Bouke commented on .travis.yml:
```diff
@@ -1,26 +1,39 @@
 language: python
 
+python:
```
Adding this will create additional entries in the build matrix. See also [the ci build](https://travis-ci.org/Bouke/django-airbrake/builds/350670567?utm_source=github_status&utm_medium=notification). No need to add this list of python versions, and control the build matrix using the `matrix:` field.

Bouke commented on setup.cfg:
```diff
@@ -1,2 +1,2 @@
 [flake8]
-max-line-length = 100
+max-line-length = 119
```
Make Guido proud by staying with 100 👍 .

Bouke replied:
Can you revert this? Or does it introduce errors for existing code?

Bouke commented on setup.py:
```diff
@@ -13,7 +13,7 @@
     license='MIT',
     long_description=open('README.rst').read(),
-    install_requires=['Django>=1.8,<1.9.99'],
+    install_requires=['Django>=1.8,<=1.11.11'],
```
I used 1.9.99 to specify all version up to/excluding 1.10 (and pre-releases of 1.10). Also specifying 1.11.11 is problematic when a bugfix is released (1.11.12) which would be excluded.

As 1.8 has almost reached EOL (April 1st), we can specify 1.11 as the lower bound and not have an upper bound.

Bouke commented on tests/tests.py:
```diff
@@ -17,24 +17,7 @@
-        xsd_validate(xml)
-
-        self.assertTrue(xml_compare(etree.fromstring("""
+DJANGO_18_RESPONSE = """
```
As we're dropping support for 1.8, is this still needed?

Bouke reviewed (changes requested):
Hey man, mooie PR! 💯 Ik kwam een paar dingetjes tegen bij de review.

- Can we also have a test for Django 2.0? If the tests fail, we can mark them as expected failure in Travis, so CI doesn't mark the build as failed.
- Can you update the readme to reflect the updated list of supported Django / Python versions?

Bounder committed:
Update django compatibility level >=1.11,<2=.0.5

Bounder committed:
Added pipfile for initializing using pipenv

Bounder committed:
Updated settings to >django1.11 compatbility

Bounder committed:
Made code django 2.0 compatible

Bounder committed:
Added tox-pyenv to pipfile

Bounder commented:
I added the requested changes except for the max-line-length as discussed. Project is now Django>=1.11,<=2.0.5 compatible. Dropped support for 1.8 as the LTS was dropped this month

Bouke commented on Pipfile.lock:
```diff
@@ -0,0 +1,206 @@
+{
```
It's not common practice for libraries to ship a lockfile. Lockfiles are great for end-users who want to have a reproducible build, with specific versions of dependencies. In this case, we should not include the lockfile.

Bounder committed:
Removed pipfile.lock from repo

Bounder commented:
Removed piffle.lock from git repo as requested

Bouke commented on setup.py:
```diff
@@ -13,7 +13,7 @@
     license='MIT',
     long_description=open('README.rst').read(),
-    install_requires=['Django>=1.8,<1.9.99'],
+    install_requires=['Django>=1.11,<=2.0.5'],
```
Requiring up to 2.0.5 will cause problems once 2.0.6 or 2.1 is released. Probably better to not specify an upper bound.

Bouke commented on tests/utils.py:
```diff
@@ -37,11 +37,13 @@ def xml_compare(x1, x2, reporter=None):
         else:
             if reporter:
+                xml_compare(c1, x2.getchildren()[i], reporter=reporter)
+                # xml_compare(c1, cl2[0], reporter=reporter)
```
Why are these changes needed?

Bounder replied:
Don't know anymore, this was however how it originally was, so probably changed it to make the test run again. Maybe you can tell me why it was changed?

Bouke commented on .travis.yml:
```diff
@@ -3,24 +3,29 @@ language: python
-      env: FLAKE8=YES
+before_install:
+  - sudo apt-get -qq update
+  - sudo apt-get install -y libxml2-utils
```
Does this PR now depend on `libxml2-utils` being available? Or only for testing?

Bounder replied:
This is needed since you changed the way to validate the xml, see commit: 69dc92c5cb557a7bd09b9a9316d93d76c6d2e390

Bounder replied:
So yes, only for testing. It's not used by airbrake itself.

Bouke reviewed (changes requested):
Great PR so far. A few minor changes and I'll merge it in!

Bounder committed:
Correspond readme to tested python versions

Bounder committed:
Set flake8 line length back to 88

Making Guido proud!

Bounder commented:
Thanks for being such a PITA, reviewing this PR. This is the kind of code quality control and peer reviewing I'm missing in my day-to-day programming. Learned a lot again! Thanks.

Bounder committed:
Removed upper django version from setup.py

Bouke commented:
I removed `Pipfile` from the repo. Would a library want to provide something like that? It seems to duplicate `requirements_dev.txt` (probably a good thing, right?), but also `setup.py` (it was still listing `django<=2.0.5`). I might need to familiarize myself with 2018's Python packaging status-quo.
